flow/src/main/java/com/sequenceiq/flow/service/flowlog/FlowLogDBService.java

```diff
@@ -333,6 +334,17 @@ public List<FlowLog> findAllByFlowIdOrderByCreatedDesc(String flowId) {
         return flowLogRepository.findAllByFlowIdOrderByCreatedDesc(flowId);
     }
 
+    public boolean isFlowConfigAlreadyRun(Long id, Class<? extends FlowConfiguration<?>> flowConfigurationClass) {
```

Comment:
I apologize for one more comment: I would change the method name. I think the name isFlowConfigAlreadyRunnING expresses the goal of the method better. AlreadyRun for me means that the flow ran at some point in the past and could be finished already.

Reply:
+1